clang/lib/Sema/SemaCoroutine.cpp

```diff
@@ -358,6 +364,30 @@ static Expr *maybeTailCall(Sema &S, QualType RetType, Expr *E,
   return S.MaybeCreateExprWithCleanups(JustAddress);
 }
 
+static bool isSpecializationOfCoroutineHandle(Sema &S, QualType Ty,
+                                              SourceLocation Loc) {
+  auto *CoroutineHandleClassTemplateDecl =
+      lookupCoroutineHandleTemplate(S, Loc);
```

Comment:
Maybe it can be better to cache the coroutine handle declaration to avoid redundant looking up. But it may be fine for the number of `await_suspend` in practical.

We can send another patch for this if we want.